src/GuzzleHttp/ClientSerializerAdapter.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the CsaGuzzleBundle package
+ *
+ * (c) Charles Sarrazin <charles@sarraz.in>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code
+ */
+
+namespace Csa\Bundle\GuzzleBundle\GuzzleHttp;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7;
+use Psr\Http\Message\RequestInterface;
+
+class ClientSerializerAdapter extends Client
+{
+    /**
+     * @inherit
+     */
+    public function sendAsync(RequestInterface $request, array $options = [])
+    {
+        $options = $this->prepareDefaults($options);
+        $response = parent::sendAsync($request, $options)->wait();
+
+        return $this->processResponse($response);
+    }
+
+    /**
+     * @inherit
+     */
+    public function requestAsync($method, $uri = null, array $options = [])
+    {
+        $options = $this->prepareDefaults($options);
+        $response = parent::requestAsync($method, $uri, $options)->wait();
+
+        return $this->processResponse($response);
+    }
+
+    /**
+     * Hook into the defaults options that cannot be changed by middleware.
+     *
+     * @param array $options
+     *
+     * @return array
+     */
+    private function prepareDefaults(array $options = [])
+    {
+        $serializerAdapter = $this->getConfig('serializer_adapter');
+
+        if ((isset($options['json'])) && $serializerAdapter) {
+            $modify['body'] = Psr7\stream_for($serializerAdapter->serialize($options['json'], 'json', []));
```

Comment:
This won't work for Guzzle 4 or 5. I guess I'll cherry-pick your commits to merge them into `1.3`, and then into `master`. Ok ? :)